src/activity_handlers.cpp

```diff
@@ -1501,6 +1502,51 @@ void activity_handlers::repair_item_finish( player_activity *act, player *p )
     act->moves_left = actor->move_cost;
 }
 
+void activity_handlers::mend_item_finish( player_activity *act, player *p )
+{
+    item_location target;
+    switch( static_cast<item_location::type>( act->index ) ) {
```

Comment:
Exposing the location types seems bad. `item_location` should be handle this via serialize/deserialize functions. For example with forwarding the player_activity to them.